### DISCUSSION TITLE
About routing constant values to VDD/VSS power rails

### DISCUSSION THREAD

**Discussion by OuDret** [The-OpenROAD-Project/OpenROAD#4602] [General]
Is there any way to route logical constant 1 & 0 to Vdd & Vss power rails instead of using tie cells? Right now I'm using a cell with short circuit between output and power pins, which is really bad in terms of area.

**Comment by maliberty** [comment]
No there is no facility for that.  The supply rails are often noisy and the tie cell provides a more stable output.  What is the issue with using a tie cell?

**Reply by OuDret** [reply]
I am using an emerging electronic technology and the number of transistor that we can introduce is a strong restriction, in this way and we experience a huge difference using them or not in our designs. I guess I will have to create copies of each cell for all possible shortened inputs to replace them after synthesis.

**Reply by maliberty** [reply]
You could use one tie hi & one tie low per design if that's acceptable.

**Reply by maliberty** [reply]
You could also make a dummy tie cell whose internal implementation is just a short.

**Reply by OuDret** [reply]
Using only one tie cell per design would increase routing congestion which is critical since i only have 2 routing layers. Dummy tie cell is what I'm currently using but it take 1 stdcell site + pad_site area.

**Comment by stefanottili** [comment]
Constant propagation should be able to remove all constants unless they're connected to ram/rom/pad/analog macro's, no ?
Which synthesis tool has been used to create the netlist ?
Can yosys do constant propagation ? If so, is it on by default or does it need to be enabled ?

**Reply by maliberty** [reply]
Yosys does constant propagation in the normal flow.

**Reply by OuDret** [reply]
I have ran ```hilomap -singleton -locell LOGIC0 Z -hicell LOGIC1 Z``` in yosys synthesis.

If you don't handle it in sI have ran ```hilomap -singleton -locell LOGIC0 Z -hicell LOGIC1 Z``` in yosys synthesis.

If you don't handle it in synthesis step, constants like 1'h0 are replaced for ```_zero``` keyword when calling ```read_verilog``` .

Openroad ``insert_tie_cells``` command replaces those keywords for tie_cells.

I don't know much about yosys but I assume it can be done some constant propagation. Will look for it
